### PR TITLE
Add a password to headless server mode

### PR DESCRIPTION
**Pull request by Giantaxe**
1. Added --password=<str> to headless server mode to configure a password
2. added default_password under [network] to configure a default password for headless server mode

### CODE REVIEW

File: src/openrct2.c [hunk 1]
```diff
@@ -287,6 +288,13 @@ void openrct2_launch()
 				if (gNetworkStartPort == 0) {
 					gNetworkStartPort = gConfigNetwork.default_port;
 				}
+
+				if (strlen(gCustomPassword) == 0) {
```
Comment (IntelOrca) [inline src/openrct2.c]: use ```str_is_null_or_empty```

File: src/cmdline/RootCommands.cpp [hunk 2]
```diff
@@ -44,6 +45,7 @@ static const CommandLineOptionDefinition StandardOptions[]
 #ifndef DISABLE_NETWORK
     { CMDLINE_TYPE_INTEGER, &_port,            NAC, "port",              "port to use for hosting or joining a server"                },
 #endif
+	{ CMDLINE_TYPE_STRING,  &_password,		   NAC, "password",			 "Server password"                                            },
```
Comment (IntelOrca) [inline src/cmdline/RootCommands.cpp]: Do not use tabs in cpp files, use spaces.

**Comment by Giantaxe** [comment]
edited the tabs to spaces
changed the password length checker to str_is_null_or_empty

**Comment by IntelOrca** [comment]
You have used the wrong indent level, please be consistent with the rest of the file.

File: src/config.c [hunk 3]
```diff
@@ -253,6 +253,7 @@ config_property_definition _twitchDefinitions[] = {
 config_property_definition _networkDefinitions[] = {
 	{ offsetof(network_configuration, player_name),						"player_name",					CONFIG_VALUE_TYPE_STRING,		{.value_string = "Player" },	NULL					},
 	{ offsetof(network_configuration, default_port),					"default_port",					CONFIG_VALUE_TYPE_UINT32,		NETWORK_DEFAULT_PORT,			NULL					},
+	{ offsetof(network_configuration, default_password),				"default_password",			    CONFIG_VALUE_TYPE_STRING,       {.value_string = NULL     },   NULL                    },
```
Comment (IntelOrca) [inline src/config.c]: You have mixed spaces and tabs here, please make sure you line them up correctly.

**Comment by Giantaxe** [comment]
fixed wrong spaces, re-aligned code with other code
same for tabs in config.c, re-aligned and spaces replaced with tabs.

**Comment by pizza2004** [comment]
I wonder why he closed this...

**Pull request closed without merging**